components/rhs_comment/rhs_comment.jsx

```diff
@@ -173,7 +173,9 @@ class RhsComment extends React.PureComponent {
     };
 
     handleAlt = (e) => {
-        this.setState({alt: e.altKey});
+        if (this.state.alt !== e.altKey) {
```

Comment:
Does this have an actual effect? This doesn't seem like it should do anything since this is a PureComponent and changing calling `setState` without making any changes wouldn't cause a re-render

Reply:
Ya, it has an actual effect. It does not update the component but it kicks in setState triggering react reconciliation algorithm affecting performance

Without:
![Jan-18-2020 01-14-19](https://user-images.githubusercontent.com/4973621/72635677-f0b93e80-398f-11ea-98cd-4e53d6314d68.gif)

with the change:
![Jan-18-2020 01-14-19](https://user-images.githubusercontent.com/4973621/72635693-fd3d9700-398f-11ea-81b3-8f1e4d605c5c.gif)

As you can see without the check from the timing API there are processes running and there are corresponding JS calls as well for them

Note: I am gonna make an item for dev meeting about this

Reply:
@lieut-data Drawing you attention as well as this seemed like an interesting thing about react